Fix #build_checkout_request_body

Naokimi opened the pull request:
In commit https://github.com/nebulab/solidus_viabill/pull/15/commits/7e1440811db87ad527315633791f8901f19b4a33 we changed request_body[:success_url] for gateway.success_url
which generates an error because the former returns a url with params, while the latter without.
With this commit we are reverting that change and aligning the rest of the parameters to match
in format with this one.

Naokimi committed:
Fix #build_checkout_request_body

In commit https://github.com/nebulab/solidus_viabill/pull/15/commits/7e1440811db87ad527315633791f8901f19b4a33 we changed request_body[:success_url] for gateway.success_url
which generates an error because the former returns a url with params, while the latter without.
With this commit we are reverting that change and aligning the rest of the parameters to match
in format with this one.

DanielePalombo commented on app/helpers/solidus_viabill/api/checkout_helper.rb:
```diff
@@ -33,13 +33,13 @@ def build_checkout_request_body(order, payment_method_id, frontend)
           }
         }
         request_body[:sha256check] = gateway.generate_signature(
-          gateway.api_key,
```
Is possible to add a test to prevent that this would happen again in the future?

Naokimi committed:
Add tests to check #build_checkout_request_body's result

Current tests just check key list which led to an error while refactoring. With
this commit we make sure to check the returned value so to avoid future surprises.
